### PR TITLE
[COL-1250] update response handling

### PR DESCRIPTION
**Pull request by esurat20**

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -4,7 +4,7 @@ defmodule Mixduty.MixProject do
   def project do
     [
       app: :mixduty,
-      version: "0.1.1",
+      version: "1.0.0",
```
Comment (yondermon) [inline mix.exs:7]: I'd make it `0.2.0`.

File: lib/mixduty/response.ex [hunk 2]
```diff
@@ -0,0 +1,69 @@
+defmodule Mixduty.Response do
+  require Logger
+  alias Mixduty.Error
+
```
Comment (yondermon) [inline lib/mixduty/response.ex:4]: nit:
```suggestion
  alias Jason, as: JSON

```
And use `JSON.decode()` instead of `Jason.decode()`

**Review by yondermon — APPROVED** [review]

**Review by gavrilenco — APPROVED** [review]

File: mix.exs (hunk 1 above)
Comment (esurat20) [reply mix.exs]: Why do you think that it is a minor changes?

File: lib/mixduty/response.ex [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+         body
+         |> extract_from_container(opts[:container])
+         |> Morphix.atomorphiform!()
+       )}
+
+  def put_into(%{} = map, struct_name, _opts), do: {:ok, struct(struct_name, map)}
+
+  def put_into(error, _module, _opts), do: error
+
+  def put_all_into(response, struct_name_or_map, opts \\ [])
```
Comment (Dashenok) [inline lib/mixduty/response.ex:97]: Where is this method used?

File: lib/mixduty/response.ex [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    {:error, %Error{message: "Unhandled status code: #{code}", status_code: code, cause: resp}}
+  end
+
+  def new({:error, %HTTPoison.Error{reason: reason} = error}) do
+    {:error, %Error{message: "HTTP request failed. Reason: #{inspect(reason)}", cause: error}}
+  end
+
+  def put_into(response, struct_name_or_map, opts \\ [])
+
+  def put_into({:ok, %__MODULE__{body: body}}, %{} = map, opts),
```
Comment (Dashenok) [inline lib/mixduty/response.ex:73]: Are you sure that first parameter is %{:ok, something}, Map.put returns a map https://github.com/PagerDuty/mixduty/pull/21/files#diff-c156091a837a956795e7aa94c3a8983de1f1ad3c14c18c74f2ab279e01340f83R35

File: lib/mixduty/response.ex (hunk 3 above)
Comment (esurat20) [reply lib/mixduty/response.ex:97]: It is not used yet.

File: lib/mixduty/response.ex (hunk 4 above)
Comment (esurat20) [reply lib/mixduty/response.ex:73]: We have this matching for this case
```
def put_into(%{} = map, struct_name, _opts), do: {:ok, struct(struct_name, map)}
```

**Review by Dashenok — APPROVED** [review]

**Pull request merged**